src/ui/shared/useUpdateSearchParam.ts

```diff
@@ -0,0 +1,23 @@
+import type { SetURLSearchParams } from 'react-router-dom';
+import { useSearchParams } from 'react-router-dom';
+
+type UpdateSearchParam = (
+  key: string,
+  value: string,
+  navigateOptions?: Parameters<SetURLSearchParams>[1]
+) => void;
+
+export function useUpdateSearchParam(): [URLSearchParams, UpdateSearchParam] {
+  const [params, setSearchParams] = useSearchParams();
+  const updateSearchParam = (
+    key: string,
+    value: string,
+    navigateOptions?: Parameters<typeof setSearchParams>[1]
+  ) => {
+    const newParams = new URLSearchParams(params);
+    newParams.set(key, value);
+    setSearchParams(newParams, navigateOptions);
+  };
+
+  return [params, updateSearchParam];
+}
```

Comment:
In my opinion name pattern of `use<verb>` is not a good one

Second, returning a tuple with `[state, setState]` isn't necessary for a custom hook; we can return regular object. The makes sense for `useState` because it's supposed to be used often. This hook isn't.

Third, I don't think this hook is helpful. The custom logic it contains is just a pure function. 
We should aim to reuse js code, not react-specific code.

It's better, in my opinion, to use the provided `useSearchParams` hook with a helper fn:

```
const [params, setSearchParams] = useSearchParams();
...
setSearchParams(setParam(params, key, value))
```